spec/controllers/idv/verify_info_controller_spec.rb

```diff
@@ -253,6 +255,69 @@
         end
       end
 
+      context 'when there is a threatmetrix exception' do
+        let(:review_status) { nil }
+
+        let(:idv_result) do
+          {
+            context: {
+              device_profiling_adjudication_reason: 'device_profiling_exception',
+              errors: {},
+              stages: {
+                threatmetrix: {
+                  client: nil,
+                  errors: {},
+                  exception: "Unexpected ThreatMetrix review_status value: #{review_status}",
+                  response_body: nil,
+                  review_status:,
+                  success: false,
+                  transaction_id: nil,
+                },
+              },
+            },
+            success: false,
+          }
+        end
+
+        it 'sets the review status in the idv session' do
+          get :show
+          expect(controller.idv_session.threatmetrix_review_status).to be_nil
+        end
+
+        it 'redirects to warning_url' do
+          get :show
+
+          expect(response).to redirect_to idv_session_errors_warning_url
+
+          expect(@analytics).to have_logged_event(
+            'IdV: doc auth warning visited',
+            step_name: 'verify_info',
+            remaining_submit_attempts: kind_of(Integer),
+          )
+        end
+
+        it 'logs the analytics event with the device profiling exception' do
+          get :show
+
+          expect(@analytics).to have_logged_event(
+            'IdV: doc auth verify proofing results',
+            hash_including(
+              success: false,
+              proofing_results: hash_including(
+                context: hash_including(
+                  device_profiling_adjudication_reason: 'device_profiling_exception',
+                  stages: hash_including(
+                    threatmetrix: hash_including(
+                      exception: match(/\S+/),
```

Comment:
A/N: I've only ever found 2 variants of the actual exception message sent to us, but didn't think a stricter match was really necessary as we don't inspect the exception message at all.